lib/ansible/executor/module_common.py

```diff
@@ -870,7 +871,17 @@ def _get_module_utils_remainder_parts(self, name_parts):
         return name_parts[5:]  # eg, foo.bar for ansible_collections.ns.coll.plugins.module_utils.foo.bar
 
 
-def recursive_finder(name, module_fqn, module_data, zf):
+def _make_zinfo(filename, date_time, zf=None):
```

Comment:
why not pass `zf.compression` instead of the entire object?
```suggestion
def _make_zinfo(filename, date_time, zf_compression=None):
```

Reply:
I had it this way originally, but for purposes of future expansion, I passed the entire object in case later we needed more.  But I'm not attached to doing so.

Reply:
I thought it might be a premature optimization, but if you foresee a specific use case that will definitely happen, keep it as is.